modules/administration-guide/partials/proc_building-a-vs-code-container-image-with-custom-branding.adoc

```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// branding-vs-code
+
+[id="building-a-vs-code-container-image-with-custom-branding_{context}"]
+= Building a Visual Studio Code - Open Source container image with custom branding
+
+This section describes how to build a Visual Studio Code container image with custom branding applied.
+
+
+.Prerequisites
+
+* A `branding/product.json` file with custom branding definitions.
+* Branding resources (icons, `codicon.css` file, `workbench-config.json` file) in the `branding` folder. See link:https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/devspaces-code/branding[an example].
+
+
+.Procedure
+
+. Run `branding.sh` script:
++
+[subs="+attributes,+quotes"]
+----
+$ ./branding/branding.sh
+----
+
+. Build the container image with Visual Studio Code and push the image to a container registry:
++
+[subs="+attributes,+quotes"]
+----
+$ {docker-cli} build -f build/dockerfiles/linux-musl.Dockerfile -t linux-musl-amd64 .
+$ {docker-cli} build -f build/dockerfiles/linux-libc.Dockerfile -t linux-libc-amd64 .
+$ export DOCKER_BUILDKIT=1
+$ {docker-cli} build -f build/dockerfiles/assembly.Dockerfile -t vs-code-open-source .
+$ {docker-cli} push username/vs-code-open-source:next
+----
```

Comment:
These commands are actual for the https://github.com/che-incubator/che-code repo.
They are different for the https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/devspaces-code repo.

Is it a problem? 
If so - please let me know what the true way to resolve it.
thanks in advance!

Reply:
Yes, we also need the commands for https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/devspaces-code. We can handle diverging/alternative content. We will publish both in two different repos (upstream docs and downstream docs).

Reply:
I suggest you prepare the commands for https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/devspaces-code and then let me know when you have them.

Reply:
thank you, @max-cx!
please see my comment below https://github.com/eclipse-che/che-docs/pull/2467#issuecomment-1322062631

Reply:
> These commands are actual for the https://github.com/che-incubator/che-code repo.
> They are different for the https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/devspaces-code repo.
> 
> Is it a problem?
> If so - please let me know what the true way to resolve it.
> thanks in advance!

I'm waiting for a confirmation from Nick and Mario.

Reply:
@RomanNikitenko and @nickboldt
The current situation is such that the following step is only available for the upstream application:
```
. Build the container image with Visual Studio Code and push the image to a container registry:
+
[subs="+attributes,+quotes"]
----
$ {docker-cli} build -f build/dockerfiles/linux-musl.Dockerfile -t linux-musl-amd64 .
$ {docker-cli} build -f build/dockerfiles/linux-libc.Dockerfile -t linux-libc-amd64 .
$ export DOCKER_BUILDKIT=1
$ {docker-cli} build -f build/dockerfiles/assembly.Dockerfile -t vs-code-open-source .
$ {docker-cli} push username/vs-code-open-source:next
----
```
And this one step blocks us from merging the whole page into the docs (because we don't want upstream-only stuff as-is to be published in the middle of the downstream docs).
Here are two options for you to consider to move forward on this:

Either

* Take more time and come up with the downstream alternatives for the above commands (within the one procedure step above).

Or

* Keep this PR open until there's some change after which this PR is updated.